components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/LoginEventHookHandler.java

```diff
@@ -0,0 +1,224 @@
+
+        EventData eventData = eventHookHandlerUtils.buildEventDataProvider(event);
+
+        if (eventData.getAuthenticationContext().isPassiveAuthenticate()) {
+            return;
+        }
+
+        //TODO: Add the implementation to read the Event Schema Type from the Tenant Configuration
+        LoginEventPayloadBuilder payloadBuilder = PayloadBuilderFactory
+                .getLoginEventPayloadBuilder(EVENT_SCHEMA_TYPE_WSO2);
+        EventAttribute loginEventPublisherConfig = getLoginEventPublisherConfigForTenant(
+                eventData.getAuthenticationContext().getLoginTenantDomain(), event.getEventName());
+        EventPayload eventPayload;
+        String eventUri;
+
+        if (IdentityEventConstants.EventName.AUTHENTICATION_SUCCESS.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationSuccessEvent(eventData);
+            eventUri = eventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_SUCCESS_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+            eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        } else if (IdentityEventConstants.EventName.AUTHENTICATION_STEP_FAILURE.name().equals(event.getEventName()) &&
+                loginEventPublisherConfig.isPublishEnabled()) {
+            eventPayload = payloadBuilder.buildAuthenticationFailedEvent(eventData);
+            eventUri = eventHookHandlerUtils.getEventUri(Constants.EventHandlerKey.LOGIN_FAILED_EVENT);
+            String tenantDomain = eventData.getAuthenticationContext().getLoginTenantDomain();
+            SecurityEventTokenPayload securityEventTokenPayload = eventHookHandlerUtils.buildSecurityEventToken(eventPayload, eventUri);
+            eventHookHandlerUtils.publishEventPayload(securityEventTokenPayload, tenantDomain, eventUri);
+        }
+    }
+
+    private EventAttribute getLoginEventPublisherConfigForTenant(String tenantDomain, String eventName) {
+
+        if (StringUtils.isEmpty(tenantDomain) || MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            return new EventAttribute();
+        }
+
+        try {
+            Condition condition = createPublisherConfigFilterCondition();
+            Resources publisherConfigResource = EventHookHandlerDataHolder.getInstance().getConfigurationManager()
+                    .getTenantResources(tenantDomain, condition);
+
+            return extractEventAttribute(publisherConfigResource, eventName);
+        } catch (ConfigurationManagementException | IdentityEventException e) {
+            log.debug("Error while retrieving event publisher configuration for tenant.", e);
```

Comment:
This is a real error so log let's log as an error ?

Reply:
Also indicate more context which denotes that it fails when handling the a particular event (eventname) .
Also while going through basically if this step fails event publishing fails, so it rather make more sense to break the flow which avoids handling this specific event with a clear error

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/39adc5d3ada05a48388b42bb7303c9c2498322d2